lightning/src/ln/channelmanager.rs

```diff
@@ -3086,6 +3098,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					msg: update
 				});
 			}
+			self.pending_events.lock().unwrap().push(events::Event::ChannelClosed { channel_id: msg.channel_id,  err: ClosureDescriptor::CounterpartyInitiated });
```

Comment:
We shouldn't generate the event here *and* in `shutdown` handling as that would be duplicative. I would think `shutdown` handling is probably the right one given the channel is then unusable, but that could lead to duplicate cases as well if we force-close later after closing_signed, so maybe just in `internal_closing_signed`, below.